### PR TITLE
[basic-response-refactoring] refactored

### PR DESCRIPTION
**Pull request by Valve**

### CODE REVIEW

**Comment by drnluz** [comment]
@Valve The specs are broken

File: lib/solr/response/solr_error.rb [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+          super(code: nil, message: nil)
+          freeze
+        end
+
+        def inspect
+          'SolrError: None'
+        end
+      end
+
+      def self.none()
```
Comment (drnluz) [inline lib/solr/response/solr_error.rb]: Usually I don't add `()` to methods without params. Just a code style choice, it's up to you.

File: lib/solr/response.rb [hunk 2]
```diff
@@ -1,47 +1,49 @@
...
+      !ok?
     end
 
-    def initialize(documents:, available_facets: [], spellcheck: Solr::Response::Spellcheck.empty)
-      @documents = documents
-      @available_facets = available_facets
-      @spellcheck = spellcheck
+    def status
+      if header.status.zero?
+        'OK'
```
Comment (drnluz) [inline lib/solr/response.rb]: What do you think about moving this `OK` string to a constant? I see that it's being used in more places.

File: lib/solr/response/solr_error.rb (hunk 1 above)
Comment (Valve) [reply lib/solr/response/solr_error.rb]: fixed, I don't know how I missed that

File: lib/solr/response.rb (hunk 2 above)
Comment (Valve) [reply lib/solr/response.rb]: yeah

**Pull request merged**